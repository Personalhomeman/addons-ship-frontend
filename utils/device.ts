import * as ReactDeviceDetect from 'react-device-detect';
import * as CompareVersions from 'compare-versions';

export const isAndroid = () => ReactDeviceDetect.isAndroid;
export const isIOS = () => ReactDeviceDetect.isIOS;
export const osVersion = () => ReactDeviceDetect.osVersion;
export const mobileModel = () => ReactDeviceDetect.mobileModel;
export const compareVersions = (firstVersion: string, secondVersion: string) => CompareVersions(firstVersion, secondVersion)
