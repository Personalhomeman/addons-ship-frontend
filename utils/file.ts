import fetch from 'node-fetch';

export const uploadFileToS3 = async (file: File | undefined, presignedUrl: string | undefined) => {
  if (!file || !presignedUrl) throw new Error('File or presignedUrl is invalid');

  const options = {
    method: 'put',
    body: file as any
  };

  return fetch(presignedUrl, options).then(resp => {
    if (!resp.ok) throw new Error(`${resp.status}: ${resp.statusText}`);
  });
};
