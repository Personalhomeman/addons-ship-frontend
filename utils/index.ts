import mapKeys from 'lodash/mapKeys';
import mapValues from 'lodash/mapValues';
import camelCase from 'lodash/camelCase';
import snakeCase from 'lodash/snakeCase';
import isPlainObject from 'lodash/isPlainObject';

export const actionTypeCreator = (prefix: string) => (actionType: TemplateStringsArray) => `${prefix}_${actionType}`;

export const camelizeKeys = <T>(input: Object): T =>
  (mapKeys(input, (_, key: string) => camelCase(key)) as unknown) as T;

export const camelizeKeysDeep = <T>(input: Object): T =>
  mapValues(camelizeKeys(input), (value: any) => (isPlainObject(value) ? camelizeKeysDeep(value) : value)) as T;

export const snakifyKeys = <T>(input: Object): T =>
  (mapKeys(input, (_, key: string) => snakeCase(key)) as unknown) as T;

export const snakifyKeysDeep = <T>(input: Object): T =>
  mapValues(snakifyKeys(input), (value: any) => (isPlainObject(value) ? snakifyKeysDeep(value) : value)) as T;
