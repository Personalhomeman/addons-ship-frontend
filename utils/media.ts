import { useState, useEffect } from 'react';

type Handler = (e: MediaQueryListEvent) => void;

const getListeners = (queries: string[], handler: Handler) =>
  queries.map(query => {
    const listener = window.matchMedia(query);

    listener.addListener(handler);

    return listener;
  });

export const mediaQuery = (...rules: string[]) => {
  const mediaRules = rules.map(r => `(min-width: ${r})`);

  const stateHandlers = rules.map(() => useState(false));

  const getStateHandler = (media: string) => {
    const idx = mediaRules.indexOf(media);

    return stateHandlers[idx];
  };

  useEffect(() => {
    const handler = ({ matches, media }: MediaQueryListEvent) => {
      const [, set] = getStateHandler(media);

      set(matches);
    };

    const listeners = getListeners(mediaRules, handler);

    listeners.forEach(({ media, matches }) => {
      const [, set] = getStateHandler(media);
      set(matches);
    });

    return () => listeners.forEach(l => l.removeListener(handler));
  });

  return stateHandlers.map(([isMatch]) => isMatch);
};
