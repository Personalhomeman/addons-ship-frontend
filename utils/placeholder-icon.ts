import { placeholderIcons } from '@/config';
import { ProjectType } from '@/models/app';

export const getIcon = (appIcon?: string, projectType?: ProjectType) => {
  let icon = appIcon;

  if (!icon) {
    icon = placeholderIcons[projectType!] || placeholderIcons.other;
  }

  const suffix = Math.random()
    .toString(16)
    .replace(/^0\./, '');

  return `${icon}?${suffix}`;
};
