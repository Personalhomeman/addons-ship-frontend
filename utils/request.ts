import fetch, { RequestInit } from 'node-fetch';

const getHeaders = (token: string) => ({
  Authorization: token
});

interface RequestParams extends RequestInit {
  url: string;
  token: string;
}

export const request = ({ url, method, token, body }: RequestParams) =>
  fetch(url, {
    method,
    headers: getHeaders(token),
    body
  });

export const get = (url: string, token: string) => request({ method: 'get', url, token });
export const del = (url: string, token: string) => request({ method: 'delete', url, token });
export const patch = (url: string, token: string) => request({ method: 'patch', url, token });
export const post = (url: string, token: string, body: RequestInit['body']) =>
  request({ method: 'post', url, token, body });
export const put = (url: string, token: string, body: RequestInit['body']) =>
  request({ method: 'put', url, token, body });
