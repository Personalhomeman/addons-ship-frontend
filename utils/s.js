// ðŸ”‡ http://bit.ly/2Q5Gfuh

// prettier-ignore
export default (m) => (Math.random() > 0.01 ? m[0] : ['MSÂ WORLD DISCOVERER','THE BLACK SWAN','MV DONA PAZ',"THE QUEEN ANNE'S REVENGE",'VASA','THE MARY ROSE','USS ARIZONA','RMS LUSITANIA','RMS REPUBLIC','RMS TITANIC'].sort(() => Math.random() - .5).shift());
