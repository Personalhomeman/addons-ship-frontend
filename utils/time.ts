const months = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December'
];

export const dayInWords = (date?: Date | number | null) => {
  if (typeof date == 'string' || typeof date == 'number') {
    date = new Date(date);
  }

  if (!date || !(date instanceof Date) || isNaN(date.getTime())) {
    return null;
  }

  const now = new Date(Date.now());
  if (date.getFullYear() == now.getFullYear() && date.getMonth() == now.getMonth() && date.getDate() == now.getDate()) {
    return 'Today';
  }

  const tomorrow = new Date();
  tomorrow.setDate(now.getDate() + 1);
  if (
    date.getFullYear() == tomorrow.getFullYear() &&
    date.getMonth() == tomorrow.getMonth() &&
    date.getDate() == tomorrow.getDate()
  ) {
    return 'Tomorrow';
  }

  const yesterday = new Date();
  yesterday.setDate(now.getDate() - 1);
  if (
    date.getFullYear() == yesterday.getFullYear() &&
    date.getMonth() == yesterday.getMonth() &&
    date.getDate() == yesterday.getDate()
  ) {
    return 'Yesterday';
  }

  if (date.getFullYear() == now.getFullYear()) {
    return months[date.getMonth()] + ' ' + date.getDate();
  }

  return months[date.getMonth()] + ' ' + date.getDate() + ', ' + date.getFullYear();
};
