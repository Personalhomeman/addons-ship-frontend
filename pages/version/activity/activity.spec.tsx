jest.mock('@/utils/media');
jest.mock('@/utils/device');
jest.mock('@/ducks/appVersion');
jest.mock('@/services/settings');

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { mediaQuery } from '@/utils/media';
import { Props as ActivityProps, Activity } from '.';
import ActivityView from './view';
import {
  mockAppVersionEvents,
  mockFailedAppVersionEvent,
  mockFinishedAppVersionEvent,
  mockInProgressAppVersionEvent,
  mockAppVersion
} from '@/mocks';

describe('ActivityView', () => {
  const defaultProps = {
    appVersionEvents: mockAppVersionEvents,
    isLoading: false
  };

  describe('when viewed on desktop', () => {
    beforeAll(() => {
      (mediaQuery as jest.Mock).mockReturnValue([true]);
    });

    it('renders the activity view correctly', () => {
      const tree = toJSON(shallow(<ActivityView {...defaultProps} />));
      expect(tree).toMatchSnapshot();
    });

    it('renders the activity view correctly while loading', () => {
      const tree = toJSON(shallow(<ActivityView {...defaultProps} />));
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when viewed on mobile', () => {
    beforeAll(() => {
      (mediaQuery as jest.Mock).mockReturnValue([false]);
    });

    it('renders the activity view correctly', () => {
      const tree = toJSON(shallow(<ActivityView {...defaultProps} />));
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when a test failed', () => {
    it('renders entry with download log button', () => {
      const tree = toJSON(shallow(<ActivityView isLoading={false} appVersionEvents={[mockFailedAppVersionEvent]} />));
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when a test is successful', () => {
    it('renders entry without download log button', () => {
      const tree = toJSON(shallow(<ActivityView isLoading={false} appVersionEvents={[mockFinishedAppVersionEvent]} />));
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when a test is in progress', () => {
    it('renders entry without download log button', () => {
      const tree = toJSON(
        shallow(<ActivityView isLoading={false} appVersionEvents={[mockInProgressAppVersionEvent]} />)
      );
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('Activity', () => {
  const defaultProps: ActivityProps = {
    appSlug: 'an-app-slug',
    isLoading: false,
    appVersionEvents: mockAppVersionEvents,
    versionId: mockAppVersion.id,
    fetchAppVersionEvents: jest.fn() as any,
    cancelPollPublishStatus: jest.fn() as any,
    startPollPublishStatus: jest.fn() as any
  };

  beforeEach(() => {
    const { startPollPublishStatus, cancelPollPublishStatus, fetchAppVersionEvents } = defaultProps;

    (([startPollPublishStatus, cancelPollPublishStatus, fetchAppVersionEvents] as any) as jest.Mock[]).forEach(mock =>
      mock.mockReset()
    );
  });

  it('renders correctly', () => {
    (mediaQuery as jest.Mock).mockReturnValue([true]);
    const tree = toJSON(shallow(<Activity {...defaultProps} />));
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    (mediaQuery as jest.Mock).mockReturnValue([true]);
    const tree = toJSON(shallow(<Activity {...defaultProps} isLoading />));
    expect(tree).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('starts polling, fetches events', () => {
      const { appSlug, versionId, startPollPublishStatus } = defaultProps;
      shallow(<Activity {...defaultProps} />);

      expect(startPollPublishStatus).toHaveBeenCalledWith({ appSlug, id: versionId });
    });
  });

  it('stops polling', () => {
    const { cancelPollPublishStatus } = defaultProps;
    const wrapper = shallow(<Activity {...defaultProps} />);

    wrapper.unmount();
    expect(cancelPollPublishStatus).toHaveBeenCalled();
  });
});
