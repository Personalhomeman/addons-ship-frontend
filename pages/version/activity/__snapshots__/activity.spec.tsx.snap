// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity renders correctly 1`] = `
<_default
  appVersionEvents={
    Array [
      Object {
        "createdAt": 2018-09-20T12:42:31.000Z,
        "createdAtTimestamp": 1537447351000,
        "id": "1234",
        "logDownloadUrl": "https://www.bitrise.io/assets/svg/logo-bitrise.svg",
        "status": "success",
        "text": "Event finished successfully!",
        "updatedAt": 2018-09-21T12:42:31.000Z,
        "updatedAtTimestamp": 1537533751000,
      },
      Object {
        "createdAt": 2018-09-22T12:42:31.000Z,
        "createdAtTimestamp": 1537620151000,
        "id": "1234",
        "logDownloadUrl": "https://www.bitrise.io/assets/svg/logo-bitrise.svg",
        "status": "in_progress",
        "text": "Doing something...",
        "updatedAt": 2018-09-25T12:42:31.000Z,
        "updatedAtTimestamp": 1537879351000,
      },
      Object {
        "createdAt": 2018-09-26T12:42:31.000Z,
        "createdAtTimestamp": 1537965751000,
        "id": "1234",
        "logDownloadUrl": "https://www.bitrise.io/assets/svg/logo-bitrise.svg",
        "status": "failed",
        "text": "An error occured.",
        "updatedAt": 2018-09-27T12:42:31.000Z,
        "updatedAtTimestamp": 1538052151000,
      },
    ]
  }
  isLoading={false}
/>
`;

exports[`Activity renders correctly when loading 1`] = `
<_default
  appVersionEvents={
    Array [
      Object {
        "createdAt": 2018-09-20T12:42:31.000Z,
        "createdAtTimestamp": 1537447351000,
        "id": "1234",
        "logDownloadUrl": "https://www.bitrise.io/assets/svg/logo-bitrise.svg",
        "status": "success",
        "text": "Event finished successfully!",
        "updatedAt": 2018-09-21T12:42:31.000Z,
        "updatedAtTimestamp": 1537533751000,
      },
      Object {
        "createdAt": 2018-09-22T12:42:31.000Z,
        "createdAtTimestamp": 1537620151000,
        "id": "1234",
        "logDownloadUrl": "https://www.bitrise.io/assets/svg/logo-bitrise.svg",
        "status": "in_progress",
        "text": "Doing something...",
        "updatedAt": 2018-09-25T12:42:31.000Z,
        "updatedAtTimestamp": 1537879351000,
      },
      Object {
        "createdAt": 2018-09-26T12:42:31.000Z,
        "createdAtTimestamp": 1537965751000,
        "id": "1234",
        "logDownloadUrl": "https://www.bitrise.io/assets/svg/logo-bitrise.svg",
        "status": "failed",
        "text": "An error occured.",
        "updatedAt": 2018-09-27T12:42:31.000Z,
        "updatedAtTimestamp": 1538052151000,
      },
    ]
  }
  isLoading={true}
/>
`;

exports[`ActivityView when a test failed renders entry with download log button 1`] = `
<Base
  paddingVertical="x8"
>
  <Table
    type="flat"
  >
    <TableBody>
      <TableRow
        key="0"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <TableCell
          style={
            Object {
              "display": "block",
            }
          }
          title="2018-09-26 12:42:31"
        >
          <Flex
            alignChildrenHorizontal="between"
            alignChildrenVertical="end"
            direction="horizontal"
            wrap={true}
          >
            <Base>
              <Base>
                September 26, 2018
                 at 
                12:42:31
              </Base>
              <Base>
                <Flex
                  alignChildrenVertical="middle"
                  direction="horizontal"
                  gap="x4"
                  height="2rem"
                >
                  <Icon
                    color="red-3"
                    name="CloseSmall"
                    size="1.5rem"
                  />
                  <Text
                    color="red-3"
                    size="x3"
                  >
                    An error occured.
                  </Text>
                </Flex>
              </Base>
            </Base>
            <a
              href="https://www.bitrise.io/assets/svg/logo-bitrise.svg"
            >
              <Button
                level="secondary"
                size="small"
              >
                <Icon
                  color="grape-4"
                  name="Download"
                  size="1.5rem"
                />
                Download Build Log
              </Button>
            </a>
          </Flex>
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</Base>
`;

exports[`ActivityView when a test is in progress renders entry without download log button 1`] = `
<Base
  paddingVertical="x8"
>
  <Table
    type="flat"
  >
    <TableBody>
      <TableRow
        key="0"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <TableCell
          style={
            Object {
              "display": "block",
            }
          }
          title="2018-09-22 12:42:31"
        >
          <Flex
            alignChildrenHorizontal="between"
            alignChildrenVertical="end"
            direction="horizontal"
            wrap={true}
          >
            <Base>
              <Base>
                September 22, 2018
                 at 
                12:42:31
              </Base>
              <Base>
                <Flex
                  alignChildrenVertical="middle"
                  direction="horizontal"
                  gap="x4"
                  height="2rem"
                >
                  <Icon
                    color="green-4"
                    name="Tick"
                    size="1.5rem"
                  />
                  <Text
                    color="grape-4"
                    size="x3"
                  >
                    Doing something...
                  </Text>
                </Flex>
              </Base>
            </Base>
          </Flex>
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</Base>
`;

exports[`ActivityView when a test is successful renders entry without download log button 1`] = `
<Base
  paddingVertical="x8"
>
  <Table
    type="flat"
  >
    <TableBody>
      <TableRow
        key="0"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <TableCell
          style={
            Object {
              "display": "block",
            }
          }
          title="2018-09-20 12:42:31"
        >
          <Flex
            alignChildrenHorizontal="between"
            alignChildrenVertical="end"
            direction="horizontal"
            wrap={true}
          >
            <Base>
              <Base>
                September 20, 2018
                 at 
                12:42:31
              </Base>
              <Base>
                <Flex
                  alignChildrenVertical="middle"
                  direction="horizontal"
                  gap="x4"
                  height="2rem"
                >
                  <Icon
                    color="green-4"
                    name="Tick"
                    size="1.5rem"
                  />
                  <Text
                    color="grape-4"
                    size="x3"
                  >
                    Event finished successfully!
                  </Text>
                </Flex>
              </Base>
            </Base>
          </Flex>
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</Base>
`;

exports[`ActivityView when viewed on desktop renders the activity view correctly 1`] = `
<Base
  paddingVertical="x8"
>
  <Table
    type="flat"
  >
    <TableHeader>
      <TableHeaderRow>
        <TableHeaderCell>
          Date
        </TableHeaderCell>
        <TableHeaderCell
          colSpan="2"
        >
          Activity
        </TableHeaderCell>
      </TableHeaderRow>
    </TableHeader>
    <TableBody>
      <TableRow
        key="0"
      >
        <TableCell
          title="2018-09-20 12:42:31"
        >
          September 20, 2018
           at 
          12:42:31
        </TableCell>
        <TableCell>
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x4"
            height="2rem"
          >
            <Icon
              color="green-4"
              name="Tick"
              size="1.5rem"
            />
            <Text
              color="grape-4"
              size="x3"
            >
              Event finished successfully!
            </Text>
          </Flex>
        </TableCell>
        <TableCell
          shrink={true}
        />
      </TableRow>
      <TableRow
        key="1"
      >
        <TableCell
          title="2018-09-22 12:42:31"
        >
          September 22, 2018
           at 
          12:42:31
        </TableCell>
        <TableCell>
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x4"
            height="2rem"
          >
            <Icon
              color="green-4"
              name="Tick"
              size="1.5rem"
            />
            <Text
              color="grape-4"
              size="x3"
            >
              Doing something...
            </Text>
          </Flex>
        </TableCell>
        <TableCell
          shrink={true}
        />
      </TableRow>
      <TableRow
        key="2"
      >
        <TableCell
          title="2018-09-26 12:42:31"
        >
          September 26, 2018
           at 
          12:42:31
        </TableCell>
        <TableCell>
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x4"
            height="2rem"
          >
            <Icon
              color="red-3"
              name="CloseSmall"
              size="1.5rem"
            />
            <Text
              color="red-3"
              size="x3"
            >
              An error occured.
            </Text>
          </Flex>
        </TableCell>
        <TableCell
          shrink={true}
        >
          <a
            href="https://www.bitrise.io/assets/svg/logo-bitrise.svg"
          >
            <Button
              level="secondary"
              size="small"
            >
              <Icon
                color="grape-4"
                name="Download"
                size="1.5rem"
              />
              Download Build Log
            </Button>
          </a>
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</Base>
`;

exports[`ActivityView when viewed on desktop renders the activity view correctly while loading 1`] = `
<Base
  paddingVertical="x8"
>
  <Table
    type="flat"
  >
    <TableHeader>
      <TableHeaderRow>
        <TableHeaderCell>
          Date
        </TableHeaderCell>
        <TableHeaderCell
          colSpan="2"
        >
          Activity
        </TableHeaderCell>
      </TableHeaderRow>
    </TableHeader>
    <TableBody>
      <TableRow
        key="0"
      >
        <TableCell
          title="2018-09-20 12:42:31"
        >
          September 20, 2018
           at 
          12:42:31
        </TableCell>
        <TableCell>
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x4"
            height="2rem"
          >
            <Icon
              color="green-4"
              name="Tick"
              size="1.5rem"
            />
            <Text
              color="grape-4"
              size="x3"
            >
              Event finished successfully!
            </Text>
          </Flex>
        </TableCell>
        <TableCell
          shrink={true}
        />
      </TableRow>
      <TableRow
        key="1"
      >
        <TableCell
          title="2018-09-22 12:42:31"
        >
          September 22, 2018
           at 
          12:42:31
        </TableCell>
        <TableCell>
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x4"
            height="2rem"
          >
            <Icon
              color="green-4"
              name="Tick"
              size="1.5rem"
            />
            <Text
              color="grape-4"
              size="x3"
            >
              Doing something...
            </Text>
          </Flex>
        </TableCell>
        <TableCell
          shrink={true}
        />
      </TableRow>
      <TableRow
        key="2"
      >
        <TableCell
          title="2018-09-26 12:42:31"
        >
          September 26, 2018
           at 
          12:42:31
        </TableCell>
        <TableCell>
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x4"
            height="2rem"
          >
            <Icon
              color="red-3"
              name="CloseSmall"
              size="1.5rem"
            />
            <Text
              color="red-3"
              size="x3"
            >
              An error occured.
            </Text>
          </Flex>
        </TableCell>
        <TableCell
          shrink={true}
        >
          <a
            href="https://www.bitrise.io/assets/svg/logo-bitrise.svg"
          >
            <Button
              level="secondary"
              size="small"
            >
              <Icon
                color="grape-4"
                name="Download"
                size="1.5rem"
              />
              Download Build Log
            </Button>
          </a>
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</Base>
`;

exports[`ActivityView when viewed on mobile renders the activity view correctly 1`] = `
<Base
  paddingVertical="x8"
>
  <Table
    type="flat"
  >
    <TableBody>
      <TableRow
        key="0"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <TableCell
          style={
            Object {
              "display": "block",
            }
          }
          title="2018-09-20 12:42:31"
        >
          <Flex
            alignChildrenHorizontal="between"
            alignChildrenVertical="end"
            direction="horizontal"
            wrap={true}
          >
            <Base>
              <Base>
                September 20, 2018
                 at 
                12:42:31
              </Base>
              <Base>
                <Flex
                  alignChildrenVertical="middle"
                  direction="horizontal"
                  gap="x4"
                  height="2rem"
                >
                  <Icon
                    color="green-4"
                    name="Tick"
                    size="1.5rem"
                  />
                  <Text
                    color="grape-4"
                    size="x3"
                  >
                    Event finished successfully!
                  </Text>
                </Flex>
              </Base>
            </Base>
          </Flex>
        </TableCell>
      </TableRow>
      <TableRow
        key="1"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <TableCell
          style={
            Object {
              "display": "block",
            }
          }
          title="2018-09-22 12:42:31"
        >
          <Flex
            alignChildrenHorizontal="between"
            alignChildrenVertical="end"
            direction="horizontal"
            wrap={true}
          >
            <Base>
              <Base>
                September 22, 2018
                 at 
                12:42:31
              </Base>
              <Base>
                <Flex
                  alignChildrenVertical="middle"
                  direction="horizontal"
                  gap="x4"
                  height="2rem"
                >
                  <Icon
                    color="green-4"
                    name="Tick"
                    size="1.5rem"
                  />
                  <Text
                    color="grape-4"
                    size="x3"
                  >
                    Doing something...
                  </Text>
                </Flex>
              </Base>
            </Base>
          </Flex>
        </TableCell>
      </TableRow>
      <TableRow
        key="2"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <TableCell
          style={
            Object {
              "display": "block",
            }
          }
          title="2018-09-26 12:42:31"
        >
          <Flex
            alignChildrenHorizontal="between"
            alignChildrenVertical="end"
            direction="horizontal"
            wrap={true}
          >
            <Base>
              <Base>
                September 26, 2018
                 at 
                12:42:31
              </Base>
              <Base>
                <Flex
                  alignChildrenVertical="middle"
                  direction="horizontal"
                  gap="x4"
                  height="2rem"
                >
                  <Icon
                    color="red-3"
                    name="CloseSmall"
                    size="1.5rem"
                  />
                  <Text
                    color="red-3"
                    size="x3"
                  >
                    An error occured.
                  </Text>
                </Flex>
              </Base>
            </Base>
            <a
              href="https://www.bitrise.io/assets/svg/logo-bitrise.svg"
            >
              <Button
                level="secondary"
                size="small"
              >
                <Icon
                  color="grape-4"
                  name="Download"
                  size="1.5rem"
                />
                Download Build Log
              </Button>
            </a>
          </Flex>
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</Base>
`;
